internal/impl/pinecone/output.go

```diff
@@ -0,0 +1,369 @@
+				Description("The host for the Pinecone index."),
+			service.NewStringField(poFieldAPIKey).
+				Secret().
+				Description("The Pinecone api key."),
+			service.NewStringEnumField(poFieldOp, string(operationUpdate), string(operationUpsert), string(operationDelete)).
+				Default(string(operationUpsert)).
+				Description("The operation to perform against the Pinecone index."),
+			service.NewInterpolatedStringField(poFieldNamespace).
+				Default("").
+				Advanced().
+				Description("The namespace to write to - writes to the default namespace by default."),
+			service.NewInterpolatedStringField(poFieldID).
+				Description("The ID for the index entry in Pinecone."),
+			service.NewBloblangField(poFieldVectorMapping).
+				Optional().
+				Description("The mapping to extract out the vector from the document. The result must be a floating point array. Required if not a delete operation."),
+			service.NewBloblangField(poFieldMetadataMapping).
+				Optional().
+				Description("An optional mapping of message to metadata in the Pinecone index entry."),
+		)
+}
+
+func init() {
+	err := service.RegisterBatchOutput(
+		"pinecone",
+		outputSpec(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (out service.BatchOutput, batchPol service.BatchPolicy, mif int, err error) {
+			if batchPol, err = conf.FieldBatchPolicy(poFieldBatching); err != nil {
+				return
+			}
+			if mif, err = conf.FieldMaxInFlight(); err != nil {
+				return
+			}
+			if out, err = newOutputWriter(conf, mgr); err != nil {
+				return
+			}
+			return
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+type operation string
+
+const (
+	operationUpdate operation = "update-vector"
+	operationUpsert operation = "upsert-vectors"
+	operationDelete operation = "delete-vectors"
+)
+
+type outputWriter struct {
+	client client
+	host   string
+	op     operation
+	logger *service.Logger
+
+	namespace       *service.InterpolatedString
+	id              *service.InterpolatedString
+	vectorMapping   *bloblang.Executor
+	metadataMapping *bloblang.Executor
+
+	pool sync.Pool
+}
+
+func newOutputWriter(conf *service.ParsedConfig, mgr *service.Resources) (*outputWriter, error) {
+	k, err := conf.FieldString(poFieldAPIKey)
+	if err != nil {
+		return nil, err
+	}
+	pc, err := pinecone.NewClient(pinecone.NewClientParams{
+		ApiKey: k,
+	})
+	if err != nil {
+		return nil, err
+	}
+	rawOp, err := conf.FieldString(poFieldOp)
+	if err != nil {
+		return nil, err
+	}
+	var op operation
+	switch rawOp {
+	case string(operationUpsert):
+		op = operationUpsert
+	case string(operationUpdate):
+		op = operationUpdate
+	case string(operationDelete):
+		op = operationDelete
+	default:
+		return nil, fmt.Errorf("invalid operation: %s", rawOp)
+	}
+	host, err := conf.FieldString(poFieldHost)
+	if err != nil {
+		return nil, err
+	}
+	if strings.HasPrefix(host, "https://") {
+		return nil, fmt.Errorf("host field must be a FQDN not a URL: %q (remove the https:// prefix)", host)
+	}
```

Comment:
I ran into this and I'm assuming others might. Pinecone's UI is a little confusing in this aspect, and the SDK just hangs if the https is added

<img width="666" alt="Screenshot 2024-07-17 at 3 40 49 PM" src="https://github.com/user-attachments/assets/db97b05a-0ce7-4196-bc18-2c6bdb65d0f3">